### DISCUSSION TITLE
Bitwarden Organizations and Collection Support?

### DISCUSSION THREAD

**Discussion by stramec** [basecamp/kamal#1295] [Q&A]
Hi apologies if I have missed this in the docs, but I'm trying to run version 2.3.0 on an existing app using bitwarden as the secrets manager.

Everything works if I use my personal account vault, but if I use a collection within my company's vault, kamal says the items can't be found?

Is this configuration supported? I have tried a few different approaches, but I don't seem to be able to get it to work.

example:

kamal secrets fetch --adapter bitwarden --account s.@..com --from 'Org Name/Kamal' CREDENTIALS
kamal secrets fetch --adapter bitwarden --account s.@..com --from 'Org Name/Kamal/CREDENTIALS'

If it is not directly supported like this, is there a workaround, feel like I can't be the only one with this set-up

Thanks in advance

**Comment by guidocollino** [comment]
Hi @stramec. Were you able to solve this problem? 

I'm using Bitwarden (actually [vaultwarden](https://github.com/dani-garcia/vaultwarden) but it almost the same) and we have several applications that we wanted to organize in different organizations within Vaultwarden and within each organization use items with custom fields for the secrets of each application. For example for the database user and password or a token for an API.

But we find the limitation that if we have 2 items with the same name (example: DB_PASSWORD) in different organizations the kamal secrets fetch command does not find the item. In these cases it asks you to use the item_id.

We understand that the limitation is in this [line of code](https://github.com/basecamp/kamal/blob/1547089da044159a934f1ef90006ae10e599d687/lib/kamal/secrets/adapters/bitwarden.rb#L27), because it does not allow sending more parameters to the get item command. One way to solve it in our case would be to be able to use the parameter --organizationid.
